### PR TITLE
chore(data-context): update runSpec mutation to support absolute spec path

### PR DESCRIPTION
**Pull request by jordanpowell88**
### Additional details
While working through the implementation on [this issue](https://github.com/cypress-io/cypress-internal/issues/611) I ran into a limitation with the current `runSpec` mutation that was implemented in [this ticket](https://github.com/cypress-io/cypress/pull/26782). We need the mutation to work by passing in an absolute `specPath`. This PR adds support for this

### Steps to test
Launch the app using `yarn dev` and open GraphQL. Run the following mutation using an absolute `specPath`:

```ts
runSpec (specPath: "/Users/blah/Desktop/application/cypress/e2e/1-getting-started/todo.cy.js") {
    __typename
    ... on RunSpecResponse {
      testingType
      browser {
        id
        name
      }
      spec {
        id
        name
      }
    }
    ... on RunSpecError {
      code
      detailMessage
    }
  }
```

### How has the user experience changed?
n/a

### PR Tasks

- [X] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/data-context/src/actions/ProjectActions.ts [hunk 1]
```diff
@@ -512,26 +512,27 @@ export class ProjectActions {
 
       let targetTestingType: TestingType
 
+      // Get relative path from the specPath to determine which testing type from the specPattern
+      const relativeSpecPath = this.ctx.path.normalize(specPath.substring(this.ctx.currentProject.length + 1, specPath.length))
```
Comment (lmiller1990) [inline packages/data-context/src/actions/ProjectActions.ts]: I think we can just use `path.relative` here.

```js
// eg...
path.relative("/a/b", "/a/b/c/d.js") //=> 'c/d.js'
```

So in this case it'd be:

```js
path.relative(this.ctx.currentProject,specPath)
```

This way, you won't need to do the calculations to get the correct characters etc.

Quick test:


```js
> curr
'/Users/lachlanmiller/code/work/cypress2/packages/launchpad'
> spec
'/Users/lachlanmiller/code/work/cypress2/packages/launchpad/src/setup/LaunchpadHeader.vue'
>
> path.relative(curr, spec)
'src/setup/LaunchpadHeader.vue'
```

**Review by lmiller1990 — COMMENTED** [review]
Looks good - testing it now, one question about how we get the path.

File: packages/data-context/src/actions/ProjectActions.ts (hunk 1 above)
Comment (jordanpowell88) [reply packages/data-context/src/actions/ProjectActions.ts]: Ahhh, yes this was what I was trying to do. Still learning some of these node best practices. Thanks!

**Review by astone123 — CHANGES_REQUESTED** [review]
Can we update the description of the GraphQL argument for the mutation to say that we're expecting an absolute path?

https://github.com/cypress-io/cypress/blob/45f4b2f40a21bbc869adc94b9f6d79c234f25fd0/packages/graphql/src/schemaTypes/objectTypes/gql-Mutation.ts#L640-L642

**Review by astone123 — APPROVED** [review]
Looks good 👍🏻

**Review by lmiller1990 — APPROVED** [review]

**Comment by lmiller1990** [comment]
Tested it out, works as expected 💯

**Pull request merged**